fog/api/proto/view.proto

```diff
@@ -268,4 +268,8 @@ message TxOutRecord {
     bytes tx_out_e_memo_data = 9;
     /// The masked token id associated to the amount field in the TxOut that was recovered
     bytes tx_out_amount_masked_token_id = 10;
+    /// The masked token id associated to the v2 amount field in the TxOut that was recovered
+    /// This field is mutually exclusive with `tx_out_amount_masked_token_id`. It's presense indicates that
+    /// a MaskedAmountV2 object was serialized, rather than a V1 object.
+    bytes tx_out_amount_v2_masked_token_id = 11;
```

Comment:
Why not make this a `oneof`?

Reply:
In this case specifically, I really want to minimize the number of bytes on the wire that go into fog view, because that's sort of the most difficult ORAM in the whole project, and we are very close to another breakpoint.

Tags are stored using the "zig zag" encoding in protobuf, and I think 10 and 11 are the same number of bytes on the wire in this encoding.

I'm not sure exactly what happens on the wire if we use a one-of, but if it would require us to store another discriminant it would be bad. That's how I assumed it worked anyways, although this PR has taught me that actually each one-of possibility has its own tag. So maybe it would be fine to use a one-of.

Reply:
I'm going to change this to oneof as suggested, thanks :)